### ISSUE TITLE
最新commit b516a80 的v2成功运行记录

### ISSUE THREAD

**Issue by e1732a364fed** [Ksuriuri/index-tts-vllm#113]
commit: b516a8026b50dedbf1f0c658d0a66d67da5f70f8
message: Update convert_hf_format.py

这个commit 的代码实际上经过修改后，已经可以运行v2模型了，（由 https://github.com/Ksuriuri/index-tts-vllm/issues/110#issuecomment-3337165657 可知）

现记录我个人的修改方案如下，供发烧友急用, 此修改有效性仅限此commit，修改之处有点多，可能有漏掉某步。。。：

1. convert_hf_format.py 中 引用中的 model 改为 model_v2，变为 from indextts.gpt.model_v2 import UnifiedVoice
2. 进行v2的模型转换
3. 复制 api_server.py 为 api_server_v2.py, 其第21行改为 对v2的引用：`from indextts.infer_vllm_v2 import IndexTTS2 as IndexTTS`
4. api_server_v2.py 第40行 `tts.registry_speaker(speaker, audio_paths_)`  注释掉
5. 将 api_server_v2.py 100行改为  `sr, wav = await tts.infer(audio_paths, text, "", seed=seed)`
6. indextts/gpt/model_vllm_v2.py 第200行，参数 从 speech_condition_input 改成 speech_condition，
7. uv pip install descript-audiotools
8. 注释掉 indextts/infer_vllm_v2.py 中 390 - 397 行的代码
10. indextts/infer_vllm_v2.py  取消注释399行, 注释掉第400行
11.  , 431行改为 `torch.tensor(codes, device=text_tokens.device),`
12.  indextts/gpt/model_vllm_v2.py, 167 行 改为 actual_end = mel_lengths[b][-1]
13. 运行 `modelscope download facebook/w2v-bert-2.0 --local_dir ~/.cache/modelscope/hub/models/IndexTeam/IndexTTS-2/w2v-bert-2.0`
14. 之后运行发现它还会访问hf来下载另一个模型 amphion/MaskGCT , 因此运行时，国内可用 命令 `HF_ENDPOINT=https://hf-mirror.com python api_server_v2.py --model_dir ~/.cache/modelscope/hub/models/IndexTeam/IndexTTS-2 --port 11996`
15. api请求使用 /tts_url 的版本而不是 /tts，如 `curl -X POST -H "Content-Type: application/json"  -d '{"text": "hello", "audio_paths": "assets/jay_promptvn.wav"}'  http://127.0.0.1:11996/tts_url --output out.wav`

修改过程中，深感作者不易和我自己的菜，我修改的时候完全不知道我在做什么！只知能运行了。。。作者大大辛苦了！（磕头）

**Comment by e1732a364fed** [comment]
呃，我运行了新commit，还是报错退出了。这里还是写一下，回退到本issue 所需的命令为 git reset --hard b516a80

我感觉我应该fork一下？不过为了尊重作者，fork最好要慎重，能不fork就不fork

**Comment by e1732a364fed** [comment]
很高兴看到作者修复了gpt问题。感觉此issue又可以关了

**Comment by dyhuachi** [comment]
> 呃，我运行了新commit，还是报错退出了。这里还是写一下，回退到本issue 所需的命令为 git reset --hard [b516a80](https://github.com/Ksuriuri/index-tts-vllm/commit/b516a8026b50dedbf1f0c658d0a66d67da5f70f8)
> 
> 我感觉我应该fork一下？不过为了尊重作者，fork最好要慎重，能不fork就不fork
> 
> 另外，我测的结果是，给的文字越长，RTF 就越低。gpt_gen_time 对 文字长度不是很敏感，我是说，gpt_gen_time 与文字长度之间不是线性关系。
> 
> 考虑到我运行时，似乎发现程序除了运行 给出的 text 之外，还会转换一大堆 用户没有给定的文字(在model_vllm_v2.py中加了一个 print 看到的)，似乎是预设的预热？不知道有没有关系，不太懂。

我也遇到这个问题了 ，不论字数长短gpt_gen_time时间都差不多，但是没有测试过文本字数超过100以上的；torch.Size([1, 123136])
>> gpt_gen_time: 13.21 seconds
>> gpt_forward_time: 0.03 seconds
>> s2mel_time: 2.44 seconds
>> bigvgan_time: 0.24 seconds
>> Total inference time: 16.03 seconds
>> Generated audio length: 5.58 seconds
>> RTF: 2.8697
这是我在2080ti 22g 单卡上的测试结果gpu_memory_utilization给到了0.55 大概和4090给的显存一样
